In [2]:
# To support both python 2 and python 3
from __future__ import division, print_function, unicode_literals

# Common imports
import numpy as np
import os
import pandas as pd

# to make this notebook's output stable across runs
np.random.seed(42)

### Introduction to Finance


I have created finance 101 tutorials to help folks learn finance and learn to code.  These tutorials will provide a brief overview of finance prior jumping into the data. The 101s are optional, but highly recommended for beginners.
The introduction to finance 101's are designed for two audiences in mind.

1) For those new to finance
  +  I will teach you finance, python, and give you the confidence to navigate the age of Big Data.
  
2) For those currently working in finance
  +  I will teach you Python basics so that you can graduate from Excel spreadsheets to the Python notebooks!

-------------------------------------------------------------------------------------------------------------------------------

### Welcome to Finance 101: Financial Statements and Company Valuation (Dividend Discount Model)

We will cover a lot of material in this 101, so grab a coffee and let's start digging in!
Topics include:
  +  Financial Statement Overview
  +  Company Valuation Overview
  +  Company Valuation Examples
  +  Python Script and Tips


### Financial Statement Overview

First, let's define some of the basics by giving an overview of financial statements. We will be evaluating the machinery and equipment maker, Caterpillar (CAT), financials. I picked CAT because it's one of my favorite companies and it used to be headquartered about 30 minutes from my hometown.

**Income Statement:** The income statement is a summary of a business's performance over a period of time. An income statement contains revenue and expenses. It ultimately answers the question did the company make money over the specified time period. You can think of an income statement like your checking account. For example, over the last 3 months was your income higher than your expenses?

**Balance Sheet:** A balance sheet is a snapshot in time of a company's health at a given point in time that separates out assets from liabilities. An asset is anything of value such as a piece of property or a piece of machinery. In personal finance, an asset would be anything that you own outright such as a car or any investments that you may have such as 401k retirement savings. A liability is a debt that is owed by the company to another party. In personal finance, a liability would be short term credit card debt or a mortgage. The balance sheet also illustrates how a company finances its operations with debt or equity. We will discuss capital structure in another session.


-------------------------------------------------------------------------------------------------------------------------------


### Company Valuation Overview

Ok, now that we have a basic understanding of how a company organizes its finances, let's jump into how companies are valued.

There are two primary ways to evaluate a company:  discounted cash flow models (DCF) and models of relative valuation (1).  DCF models determine the value of a company based on its fundamentals such as information from the financial statements.  Relative valuation models use ratios such as Price-to-Earnings (P/E) ratio to determine valuation.  We will be focusing on a specific DCF model called the Dividend Discount Model (DDM) in this tutorial because of its simplicity and ease of use.  

DDM is a great place to start when calculating company value because it illustrates the basic principles of company valuation.  This model assumes that "A share of a company is worth the present value of all the cash flows an investor expects to pocket from it." (1)  In a very simplistic world, this is like how a bond is valued.  An investor buys a bond for an expected yield (i.e. 3%) and the bond price is determined according to this yield.  In terms of stocks, a dividend is like a bond yield because it is the cash that an investor receives from the investment.   

We will discuss two examples of the DDM calculation in this tutorial.  First, we will review the simplest form of DDM which defines a company's value based on its dividend (D) divided by a required return on equity (R). Next, we will modify that formula to include the growth of the dividend in the calculation.

Example 1 DDM Assumptions

  +  Investor holds stock forever
  +  Dividend does not grow
  +  Calculation: Value = D / R 
    +    D = Dividend
    +    R = Return on Equity
       
Example 2 DDM Assumptions
  +  Investor holds stock forever
  +  Dividend grows in line with company history
  + Calculation: Value = (D*1+g) / (R - g)
    + D = Dividend
    + R = Return on Equity
    + g = Dividend Growth Rate
 

In our example, we will assume the investor holds the stock forever.  However, in the real world this is probably not very accurate because investors change investments all the time.  Additionally, the underlying principle, (i.e. bond and stock investment) can change and the investor isn't guaranteed to receive their original investment back and companies can cut the dividend so it is not guaranteed.  Either way, the fundamentals of the dividend discount model hold true throughout all valuation models.  More specifically, a company's value is based on future cash flows given to investors.  As we will see with the next financial 101 tutorials, the art of company valuation is replacing the dividend with a better estimate of cash flows.

-------------------------------------------------------------------------------------------------------------------------------

### Company Valuation Examples

Next, we are going to execute some basic python code where we will review the Income Statement. This is the only financial statement that we need to calculate the DDM. We will be using the pandas library to import the CSV of the CAT Income Statement into a pandas dataframe. A dataframe allows us to have multiple rows and columns and is similar to looking at raw data from a CSV in Excel.

In [49]:
is_df = pd.read_csv("CAT_IS.csv") # Source: (2)
is_df.set_index("Account", inplace=True)

Next let's take a look at the Income Statement that we imported.

In [17]:
is_df

TY2016    TY2015  \
Account                                                                  
Sales of Machinery, Energy & Transportation         35773.00  44147.00   
Revenues of Financial Products                       2764.00   2864.00   
Total sales and revenues                            38537.00  47011.00   
Cost of goods sold                                  28309.00  33546.00   
Selling, general and administrative expenses         4686.00   4951.00   
Research and development expenses                    1951.00   2119.00   
Interest expense of Financial Products                596.00    587.00   
Goodwill impairment charge                            595.00      0.00   
Other operating (income) expenses                    1902.00   2023.00   
Total operating costs                               38039.00  43226.00   
Operating profit                                      498.00   3785.00   
Interest expense excluding Financial Products         505.00    507.00   
Other income (expense)                                146.00    161.00   
Consolidated profit before taxes                      139.00   3439.00   
Provision (benefit) for income taxes                  192.00    916.00   
Profit (loss) of consolidated companies               -53.00   2523.00   
Equity in profit (loss) of unconsolidated affil...     -6.00      0.00   
Net Income                                            -59.00   2523.00   
Basic Shares Outstanding (millions)                   584.30    594.30   
Dividend per Share                                      3.08      3.01   

                                                     TY2014  
Account                                                      
Sales of Machinery, Energy & Transportation         52142.0  
Revenues of Financial Products                       3042.0  
Total sales and revenues                            55184.0  
Cost of goods sold                                  40718.0  
Selling, general and administrative expenses         6529.0  
Research and development expenses                    2380.0  
Interest expense of Financial Products                624.0  
Goodwill impairment charge                              0.0  
Other operating (income) expenses                    1619.0  
Total operating costs                               51870.0  
Operating profit                                     3314.0  
Interest expense excluding Financial Products         484.0  
Other income (expense)                                322.0  
Consolidated profit before taxes                     3152.0  
Provision (benefit) for income taxes                  692.0  
Profit (loss) of consolidated companies              2460.0  
Equity in profit (loss) of unconsolidated affil...      8.0  
Net Income                                           2468.0  
Basic Shares Outstanding (millions)                   617.2  
Dividend per Share                                      2.7

Perfect the dataframe looks great and now we just need to use row 19 to get the most current dividend for our calculation of DDM.

One option would be to create a variable as follows:

D2016 = 3.08

This is equivalent to hardcoding in Excel and is bad etiquette for python as well.  In python, it's much better to take a variable from the value in the dataframe.  We can create a dividend variable from the python database below by slicing.

In [22]:
D2016 = is_df.loc["Dividend per Share","TY2016"]

In [23]:
D2016

3.0800000000000001

Ok, great.  This worked and we now have a dividend value for 2016 that we will use for our DDM valuation.

In [50]:
# In this section, I just document where I received the information from the calculate R.
# Remember consider this as given for now.

Beta = 1.29 #(3) 
Market_Risk_Prem =0.055 #(1) Chapter 7
RF = 0.0235 #(4) 10 year treasury rate

R = RF + Beta * Market_Risk_Prem
print (R)

0.09445


Now we have the Dividend and Rate of Return.  Let's calculate the share price for DDM the formula for DDM Example 1.

In [51]:
DDM_V1 = D2016 / R
print (DDM_V1)

32.6098464796


This valuation seems very low considering that CAT's stock price 52 week range is \$90.34 to \$140.44 and the current stock price is \$137 as of 11/20/2017 (3).

The major issue with the DDM example 1 calculation is that it assumes dividends will not grow.  We can clearly see from CAT's
income statement that dividends have steadily increased from 2014 to 2016.  Let's run thru the math to determine how much dividends are growing over this time period.

In [26]:
D2015 = is_df.loc["Dividend per Share","TY2015"]
D2014 = is_df.loc["Dividend per Share","TY2014"]

In [27]:
print(D2015)
print(D2014)

3.01
2.7


In [29]:
D2015_growth_rate = D2015/D2014 -1
D2016_growth_rate = D2016/D2015 -1

In [30]:
print(D2015_growth_rate)
print(D2016_growth_rate)

0.114814814815
0.0232558139535


A neat feature that we can use in python is lists.  We will create a list to store these dividend growth rates.  The list will allow us to perform mathemtical calculations in a more user friendly manner.

In [ ]:
list = [D2015_growth_rate, D2016_growth_rate]

In [36]:
list

[0.1148148148148147, 0.023255813953488413]

In [37]:
len(list)

2

In [46]:
g = sum(list) / len(list)

In [47]:
g

0.069035314384151558

We now have calculated the average dividend growth rate for CAT, so let's use it to calculate DDM for example 2.

In [48]:
DDM_V2 = (D2016*(1+g))/ (R - g)
print (DDM_V2)

129.556147893


### Conclusion

The DDM example 2 is much closer to the actual stock price and calculates that CAT is overvalued.  In fact, the DDM model will generally underestimate the value of a company because it assumes that all relevant information is taken into account by the dividend.  This is a virtually impossible assumption to achieve because a dividend is not based on perfect information. In fact, companies are incentivized to keep the dividend and a downward dividend adjustment will have very negative consequences on the stock price (1).

Additionally, there is a blind spot for the DDM because it is not able to evaluate companies that do not pay a dividend.  For example, the DDM would not be able to calculate a value for Alphabet Inc and Facebook.

For these reasons, we will move to a more advanced company valuation model called the Weighted Average Cost of Capital (WACC) in the next tutorial.  This model is like the DDM model in that it uses cash flows to determine company value.  However, the WACC model does not use the dividend as a proxy for cash flows.  The WACC model is the most used discounted cash flow model and we will get to explore the balance sheet and statement of cash flows during the tutorial.


Please upvote this tutorial if you would like to learn more about the basics of finance and company valuation.  Also, please share your comments, so that we can tailor the training to your interests!

### Works Cited:
1. The Financial Times Guide to Understanding Finance, 2nd Edition by Javier Estrada Published by Pearson Business, 2011
2. https://www.sec.gov/edgar.shtml
3. https://finance.google.com/finance?q=CAT&ei=MO4SWui5GYfSjAHDvqqQDA
4. https://ycharts.com/indicators/10_year_treasury_rate